### PR TITLE
Delegation Program Updates

### PR DESCRIPTION
**Pull request by qwadratic**

### CODE REVIEW

**Comment by qwadratic** [comment]
@barriebyron It seem to work locally but deployment pipeline failed.

File: docs/zkapps/tutorials/09-recursion.mdx [hunk 1]
```diff
@@ -66,7 +66,7 @@ To create a ZkProgram, start with the `init()` method.
 ```typescript
 const Add = ZkProgram({
   name: 'add-example',
-  publicInputType: Field,
+  publicInput: Field,
```
Comment (barriebyron) [inline docs/zkapps/tutorials/09-recursion.mdx:69]: similar feedback 

this fix is outside the scope of this PR

can you remove this file update from the PR?
One way to get this file from the main branch is:

`git checkout main docs/zkapps/o1js/recursion.mdx`

File: docs/zkapps/tutorials/09-recursion.mdx [hunk 2]
```diff
@@ -266,7 +266,7 @@ class RollupState extends Struct({
 
 const Rollup = ZkProgram({
   name: "rollup-example",
-  publicInputType: Field,
+  publicInput: Field,
```
Comment (barriebyron) [inline docs/zkapps/tutorials/09-recursion.mdx:269]: this fix is outside the scope of this PR

can you remove this file update from the PR?
One way to get this file from the main branch is:

`git checkout main docs/zkapps/o1js/recursion.mdx`

**Comment by barriebyron** [comment]
@qwadratic lmk if you have questions about my request to keep the PR in scope

Happy New Year
